src/main/kotlin/RecordProcessor.kt

```diff
@@ -50,13 +56,45 @@ class RecordProcessor() {
             throw e
         }
     }
+
+    open fun sendMessageToDlq(record: ConsumerRecord<ByteArray, ByteArray>) {
+        val body = record.value()
+        val malformedRecord = MalformedRecord(String(body), "Not a valid json")
+        try {
+            val jsonString = Klaxon().toJsonString(malformedRecord)
+            val producerRecord = ProducerRecord<ByteArray, ByteArray>(
+                dlqTopic,
+                null,
+                null,
+                record.key(),
+                jsonString.toByteArray(),
+                null
+            )
+            val metadata = DlqProducer.getInstance()?.send(producerRecord)?.get()
+            log.info("metadata topic : %s offset : %s".format(metadata?.topic(), metadata?.offset()))
+        } catch (e: Exception) {
+            log.warning(
+                ("Error while sending message to dlq : " +
+                    "key %s from topic %s with offset %s : %s").format(record.key(), record.topic(), record.offset(), e.toString()))
+            throw DlqException("Exception while sending message to DLQ " + e)
+        }
+    }
+
+    fun getObjectAsByteArray(obj: MalformedRecord): ByteArray? {
```

Comment:
Is this function used anywhere?

Reply:
Yes it is used to convert object to bytearray to place in dlq